File: src/main/java/com/gymtracker/gym/templateExercises/dto/TemplateExerciseRequest.java
package com.gymtracker.gym.templateExercises.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TemplateExerciseRequest {

    @NotNull
    private Long workoutTemplateId;
    @NotBlank
    private String exerciseName;
    @PositiveOrZero
    private Integer defaultSets;
    @PositiveOrZero
    private Integer defaultReps;
    @PositiveOrZero
    private BigDecimal defaultWeightKg;
    @PositiveOrZero
    private Integer sortOrder;
}

File: src/main/kotlin/com/gymtracker/gym/templateExercises/dto/TemplateExerciseRequest.kt
package com.gymtracker.gym.templateExercises.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.PositiveOrZero
import java.math.BigDecimal

data class TemplateExerciseRequest(
    val workoutTemplateId: Long,
    @field:NotBlank val exerciseName: String,
    @field:PositiveOrZero val defaultSets: Int?,
    @field:PositiveOrZero val defaultReps: Int?,
    @field:PositiveOrZero val defaultWeightKg: BigDecimal?,
    @field:PositiveOrZero val sortOrder: Int?,
)
